### ISSUE TITLE
##[error]Progress check failed multiple times. Aborting check sequence released

### ISSUE THREAD

**Issue by Arthurvdv** [HodorNV/ALOps#713]
**Describe the bug**
The `ALOpsExtensionAPI@1` throws an error on Publish action `Progress check failed multiple times. Aborting check sequence.`
The installation of the extension is successful, but it's the polling of the deployment status where something is going wrong.

I think it's maybe related to https://github.com/HodorNV/ALOps/issues/668?

It's an On-Premise version of Business Central
Version: BE Business Central 20.4 (Platform 20.0.44266.44359 + Application 20.4.44313.44365)

**the output**
```
2024-01-04T08:20:07.1479842Z ##[section]Starting: ALOps Extension API
2024-01-04T08:20:07.1587105Z ==============================================================================
2024-01-04T08:20:07.1587269Z Task         : ALOps Extension API
2024-01-04T08:20:07.1587482Z Description  : Get/Publish extensions with the Business Central API
2024-01-04T08:20:07.1587589Z Version      : 1.461.5079
2024-01-04T08:20:07.1587679Z Author       : Hodor
2024-01-04T08:20:07.1588425Z Help         : Get/Publish extensions with the Business Central API.
2024-01-04T08:20:07.1588533Z ==============================================================================
2024-01-04T08:20:08.2704610Z *** Validate configuration
2024-01-04T08:20:08.7377496Z *** Task Inputs:
2024-01-04T08:20:08.7430688Z 
2024-01-04T08:20:08.7493136Z name                                                                                       value
2024-01-04T08:20:08.7495559Z ----                                                                                       -----
2024-01-04T08:20:08.7495936Z usedocker                                                                                  False
2024-01-04T08:20:08.7497267Z fixed_tag                                                                                       
2024-01-04T08:20:08.7498242Z interaction                                                                              publish
2024-01-04T08:20:08.7499516Z api_endpoint                                         https://<redacted>:7048/BC20_BE_LIVE_WS/api
2024-01-04T08:20:08.7500366Z dev_endpoint                                                                                    
2024-01-04T08:20:08.7501549Z dev_schemeupdatemode                                                                 synchronize
2024-01-04T08:20:08.7502718Z apiversion                                                                                  v1.0
2024-01-04T08:20:08.7503946Z authentication                                                                  usernamepassword
2024-01-04T08:20:08.7505170Z azure_tenant_id                                                                                 
2024-01-04T08:20:08.7506247Z azure_app_client_id                                                                             
2024-01-04T08:20:08.7507283Z azure_app_client_secret                                                                         
2024-01-04T08:20:08.7508445Z azure_app_client_certificate                                                                    
2024-01-04T08:20:08.7509365Z azure_app_client_certificate_password                                                           
2024-01-04T08:20:08.7510677Z username                                                                          ***
2024-01-04T08:20:08.7511632Z password                                            ***
2024-01-04T08:20:08.7512576Z bccompany                                                                                       
2024-01-04T08:20:08.7513699Z artifact_path                                                                          D:\a\r1\a
2024-01-04T08:20:08.7514750Z artifact_filter                                                                            *.app
2024-01-04T08:20:08.7515825Z showdeploymentstatus                                                                       False
2024-01-04T08:20:08.7516946Z checksecondsdelay                                                                             30
2024-01-04T08:20:08.7517971Z maxtries                                                                                      20
2024-01-04T08:20:08.7519102Z replacepackageid                                                                           False
2024-01-04T08:20:08.7520269Z blocksymbolsonly                                                                           False
2024-01-04T08:20:08.7520383Z 
2024-01-04T08:20:08.7541268Z 
2024-01-04T08:20:08.7549816Z 
2024-01-04T08:20:08.7718420Z *** For documentation, please visit   : https://www.alops.be/documentation
2024-01-04T08:20:08.7721831Z 
2024-01-04T08:20:09.1839614Z *** ALOps License:
2024-01-04T08:20:09.1856097Z   * Licensed To: Van Roey (Collection License)
2024-01-04T08:20:09.1859941Z 
2024-01-04T08:20:09.2650725Z *** Imported required PS-Functions
2024-01-04T08:20:09.2664191Z *** Parse Task Parameters
2024-01-04T08:20:09.4044083Z *** Creating API Connector with endpoint [https://<redacted>:7048/BC20_BE_LIVE_WS/api]
2024-01-04T08:20:09.4408497Z *** API Version: v1.0
2024-01-04T08:20:09.4429581Z *** Interaction Mode: publish
2024-01-04T08:20:09.4458390Z *** API Authentication: Using Username/Password Authentication [***]
2024-01-04T08:20:09.4503412Z *** BC Authentication: Basic [***]
2024-01-04T08:20:09.4605428Z *** List Companies
2024-01-04T08:20:09.4687161Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies'
2024-01-04T08:20:09.6186859Z 
2024-01-04T08:20:09.6247537Z 
2024-01-04T08:20:09.7289751Z *** Selected Company: <redacted> [c055c5ab-4d67-ee11-9ae1-005056bd28ca]
2024-01-04T08:20:09.7392831Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensions'
2024-01-04T08:20:09.8517231Z *** Installed Extensions:
2024-01-04T08:20:09.8553960Z 
2024-01-04T08:20:09.8948492Z 
2024-01-04T08:20:09.9433056Z @odata.etag     : W/"JzIwOzE2MjQ1MjQxNTc4NzMyMjAwNzc1MTswMDsn"
2024-01-04T08:20:09.9433916Z packageId       : 1f638c00-e347-4bc8-90f3-9dda4b286115
2024-01-04T08:20:09.9434282Z id              : 73adf5b5-75bd-4110-ad76-187cf36c22fb
2024-01-04T08:20:09.9434796Z displayName     : <redacted>
2024-01-04T08:20:09.9435207Z publisher       : VanRoey | EuroSys
2024-01-04T08:20:09.9435418Z versionMajor    : 20
2024-01-04T08:20:09.9435749Z versionMinor    : 0
2024-01-04T08:20:09.9435923Z versionBuild    : 112721
2024-01-04T08:20:09.9436301Z versionRevision : 0
2024-01-04T08:20:09.9436676Z isInstalled     : True
2024-01-04T08:20:09.9437255Z publishedAs     :  PTE
2024-01-04T08:20:09.9437352Z 
2024-01-04T08:20:09.9475556Z @odata.etag     : W/"JzE5Ozg1NDQ4MDUyMTQyMTM3NDQ1MzUxOzAwOyc="
2024-01-04T08:20:09.9476134Z packageId       : 16be87a9-3437-49de-8cf6-3a71644ff90e
2024-01-04T08:20:09.9476449Z id              : d922339a-caf2-44d8-ad47-9c1eff689769
2024-01-04T08:20:09.9476886Z displayName     : <redacted>
2024-01-04T08:20:09.9477082Z publisher       : VanRoey | EuroSys
2024-01-04T08:20:09.9477428Z versionMajor    : 19
2024-01-04T08:20:09.9477612Z versionMinor    : 0
2024-01-04T08:20:09.9477992Z versionBuild    : 93568
2024-01-04T08:20:09.9478410Z versionRevision : 0
2024-01-04T08:20:09.9478882Z isInstalled     : True
2024-01-04T08:20:09.9479403Z publishedAs     : Globaal
2024-01-04T08:20:09.9642279Z 
2024-01-04T08:20:09.9642660Z 
2024-01-04T08:20:09.9644012Z 
2024-01-04T08:20:09.9649197Z 
2024-01-04T08:20:09.9787216Z *** Import NAV/BC Management DLL's [False]
2024-01-04T08:20:10.0090268Z *** ServiceTier not specified, loading default assemblies.
2024-01-04T08:20:10.0169928Z *** Default Assemblies not found. Load via BC-Artifacts
2024-01-04T08:20:10.0447994Z *** Install/Update BCContainerHelper [https://github.com/microsoft/navcontainerhelper]
2024-01-04T08:20:10.0494883Z *** Check BcContainerHelper
2024-01-04T08:21:20.3847198Z *** Install BcContainerHelper
2024-01-04T08:21:31.7447119Z *** Load BcContainerHelper
2024-01-04T08:21:32.0115144Z BcContainerHelper version 6.0.3
2024-01-04T08:21:32.5995391Z BC.HelperFunctions emits usage statistics telemetry to Microsoft
2024-01-04T08:21:35.0370812Z 
2024-01-04T08:21:35.0371580Z ModuleType Version    Name                                ExportedCommands                                             
2024-01-04T08:21:35.0372269Z ---------- -------    ----                                ----------------                                             
2024-01-04T08:21:35.0372711Z Script     6.0.3      BcContainerHelper                   {Add-FontsToBcContainer, Add-GitToAlProjectFolder, AddTele...
2024-01-04T08:21:35.0372867Z 
2024-01-04T08:21:35.0372945Z 
2024-01-04T08:21:35.0372997Z 
2024-01-04T08:21:35.0376522Z 
2024-01-04T08:21:35.0396628Z ##[command]"Get-BCArtifactUrl" 
2024-01-04T08:21:35.0396891Z Name                           Value                                                                                   
2024-01-04T08:21:35.0397222Z ----                           -----                                                                                   
2024-01-04T08:21:35.0397422Z select                         Latest                                                                                  
2024-01-04T08:21:35.0397838Z type                           OnPrem                                                                                  
2024-01-04T08:21:35.0398164Z country                        W1                                                                                      
2024-01-04T08:21:35.0398256Z 
2024-01-04T08:21:35.0398424Z 
2024-01-04T08:21:35.0398704Z  -select: Latest -type: OnPrem -country: W1
2024-01-04T08:21:37.6964303Z Downloading artifact /onprem/23.2.14098.14562/w1
2024-01-04T08:21:37.7823821Z Downloading C:\Users\VssAdministrator\AppData\Local\Temp\75f39f05-506d-461a-bb6b-b43fbb61e6bf.zip
2024-01-04T08:21:37.7861144Z Downloading using WebClient
2024-01-04T08:21:41.9150967Z Unpacking artifact to tmp folder using 7zip
2024-01-04T08:21:45.8644011Z Downloading platform artifact /onprem/23.2.14098.14562/platform
2024-01-04T08:21:45.8674233Z Downloading C:\Users\VssAdministrator\AppData\Local\Temp\3d465bfc-af8d-4283-9cbd-6b864fc1495d.zip
2024-01-04T08:21:45.8682997Z Downloading using WebClient
2024-01-04T08:22:01.4552796Z Unpacking platform artifact to tmp folder using 7zip
2024-01-04T08:22:29.6716526Z Downloading Prerequisite Components
2024-01-04T08:22:29.6890344Z Downloading C:\bcartifacts.cache\onprem\23.2.14098.14562\platform\Prerequisite Components\DotNetCore\DotNetCore.1.0.4_1.1.1-WindowsHosting.exe
2024-01-04T08:22:29.6901353Z Downloading using WebClient
2024-01-04T08:22:31.0015669Z Downloading C:\bcartifacts.cache\onprem\23.2.14098.14562\platform\Prerequisite Components\IIS URL Rewrite Module\rewrite_2.0_rtw_x64.msi
2024-01-04T08:22:31.0023701Z Downloading using WebClient
2024-01-04T08:22:31.8363876Z *** Platform: C:\bcartifacts.cache\onprem\23.2.14098.14562\platform
2024-01-04T08:22:31.8393236Z *** Localisation: C:\bcartifacts.cache\onprem\23.2.14098.14562\w1
2024-01-04T08:22:31.9141315Z *** BC Management DLL: C:\bcartifacts.cache\onprem\23.2.14098.14562\platform\ServiceTier\program files\Microsoft Dynamics NAV\230\Service\Management\Microsoft.Dynamics.Nav.Management.dll
2024-01-04T08:22:31.9240828Z *** Selected module: [C:\bcartifacts.cache\onprem\23.2.14098.14562\platform\ServiceTier\program files\Microsoft Dynamics NAV\230\Service\Management\Microsoft.Dynamics.Nav.Management.dll]
2024-01-04T08:22:31.9248079Z *** Loading module: [C:\bcartifacts.cache\onprem\23.2.14098.14562\platform\ServiceTier\program files\Microsoft Dynamics NAV\230\Service\Management\Microsoft.Dynamics.Nav.Management.dll]
2024-01-04T08:22:32.0354066Z *** BC App Management DLL: C:\bcartifacts.cache\onprem\23.2.14098.14562\platform\ServiceTier\program files\Microsoft Dynamics NAV\230\Service\Management\Microsoft.Dynamics.Nav.Apps.Management.dll
2024-01-04T08:22:32.0370821Z *** Selected module: [C:\bcartifacts.cache\onprem\23.2.14098.14562\platform\ServiceTier\program files\Microsoft Dynamics NAV\230\Service\Management\Microsoft.Dynamics.Nav.Apps.Management.dll]
2024-01-04T08:22:32.0378604Z *** Loading module: [C:\bcartifacts.cache\onprem\23.2.14098.14562\platform\ServiceTier\program files\Microsoft Dynamics NAV\230\Service\Management\Microsoft.Dynamics.Nav.Apps.Management.dll]
2024-01-04T08:22:32.0569314Z *** Publish Extension
2024-01-04T08:22:32.0577315Z   * Artifact Path  : D:\a\r1\a
2024-01-04T08:22:32.0584680Z   * Artifact Filter: *.app
2024-01-04T08:22:32.0711487Z *** Resolve App file in [D:\a\r1\a]
2024-01-04T08:22:32.0755910Z *** Resolved App File: [D:\a\r1\a\App\<redacted>\VanRoey _ EuroSys_<redacted>_20.0.120883.0_onprem_22.0_be.app].
2024-01-04T08:22:32.0774941Z *** Resolved App Path: [D:\a\r1\a\App\<redacted>\VanRoey _ EuroSys_<redacted>_20.0.120883.0_onprem_22.0_be.app]
2024-01-04T08:22:32.0919821Z *** Upload extension [D:\a\r1\a\App\<redacted>\VanRoey _ EuroSys_<redacted>_20.0.120883.0_onprem_22.0_be.app]
2024-01-04T08:22:32.0929828Z *** Get App Info
2024-01-04T08:22:32.6717990Z  * App.PackageID = 9e140f13-6d06-44ce-8b7d-67261f4d6e88
2024-01-04T08:22:32.6761599Z  * App.ID        = 73adf5b5-75bd-4110-ad76-187cf36c22fb
2024-01-04T08:22:32.6791088Z  * App.Name      = <redacted>
2024-01-04T08:22:32.6813884Z  * App.Publisher = VanRoey | EuroSys
2024-01-04T08:22:32.6836356Z  * App.Version   = 20.0.120883.0
2024-01-04T08:22:32.6893156Z *** Get Extension status
2024-01-04T08:22:32.6935746Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensions'
2024-01-04T08:22:32.8270448Z ##[command]Invoke-RestMethod -Method Patch -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionUpload(0)/content'
2024-01-04T08:22:47.1124017Z 
2024-01-04T08:22:47.1136314Z *** Poll Deployment status
2024-01-04T08:22:47.1204456Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:23:20.8391669Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:23:50.5690286Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:24:20.7144889Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:24:50.8796209Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:25:21.0767098Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:25:51.2183214Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:26:21.3770556Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:26:51.5336838Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:27:22.3502032Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:27:51.8345783Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:28:21.9866055Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:28:52.1702429Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:29:22.3384238Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:29:52.4827255Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:30:22.6477656Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:30:52.8190933Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:31:22.9614984Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:31:53.1067679Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:32:23.2511417Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://<redacted>:7048/bc20_be_live_ws/api/microsoft/automation/v1.0/companies(c055c5ab-4d67-ee11-9ae1-005056bd28ca)/extensionDeploymentStatus'
2024-01-04T08:32:23.4415755Z ##[error]Progress check failed multiple times. Aborting check sequence.
2024-01-04T08:32:23.4426328Z *** Deployment Status:
2024-01-04T08:32:23.4426523Z 
2024-01-04T08:32:23.4426583Z 
2024-01-04T08:32:23.4427136Z operationID : 5c887677-c54d-465c-a46e-d566d7f66421
2024-01-04T08:32:23.4427348Z name        : <redacted>
2024-01-04T08:32:23.4427590Z publisher   : VanRoey | EuroSys
2024-01-04T08:32:23.4427833Z appVersion  : 20.0.120883.0
2024-01-04T08:32:23.4428065Z status      : Voltooid
2024-01-04T08:32:23.4428356Z startedOn   : 2024-01-04T08:22:47.09Z
2024-01-04T08:32:23.4428517Z 
2024-01-04T08:32:23.4428573Z 
2024-01-04T08:32:23.4428625Z 
2024-01-04T08:32:23.4429583Z ##[error]App publish failed. Check the 'Extension Deployment Status' page in the Business Central client.
2024-01-04T08:32:23.4676935Z *** Cleanup VSTS Environment: True
2024-01-04T08:32:25.8161137Z ##[section]Finishing: ALOps Extension API
```

**Comment by waldo1001** [comment]
Is this consistant?

**Comment by Arthurvdv** [comment]
Unfortunately yes

![image](https://github.com/HodorNV/ALOps/assets/44637996/8a976813-7877-4a62-b547-6c023e0288c8)

**Comment by waldo1001** [comment]
The language is the issue - we now also force this webservice to English.. so should be fixed in next release.

**Comment by waldo1001** [comment]
Should have been fixed in 1.462.

**Comment by Arthurvdv** [comment]
Thanks!
